File: Emiliano_3er_sem/Java_3er_sem/Leccion1/Mundo_pc/Raton.java

package Emiliano_3er_sem.Java_3er_sem.Leccion1.Mundo_pc;

public class Raton extends DispositivoEntrada {
    private final int idRaton;
    private static int contadorRatones;

    public Raton(String tipoEntrada, String marca){
        super(tipoEntrada, marca);
        this.idRaton = ++Raton.contadorRatones;
    }

    @Override
    public String toString(){
        return "Raton{" + "IdRaton=" + idRaton + ", " + super.toString()+"}";
    }
} 

File: Emiliano-3er-sem/Python-3er-sem/Leccion14/mundo_pc/raton.py
from mundo_pc.dispositivo_entrada import Dispositivoentrada

class Raton(Dispositivoentrada):
    contador_ratones = 0
    def __init__(self, marca, tipo_entrada):
        Raton.contador_ratones += 1
        self._id_raton = Raton.contador_ratones
        super().__init__(marca, tipo_entrada)
        
    def __str__(self):
        return f"Id {self._id_raton}, Marca: {self.marca}, Tpo entrada: {self.tipo_entrada}"
    
    
#Hacemos pruebas
if __name__ == " __name__ ":
    raton1 = Raton("HP", "USB")
    print(raton1)
    raton2 = Raton("Acer", "USB")
    print(raton2)
